### PR TITLE
[zosfiles] Handle control characters in ds member names

### PR DESCRIPTION
**Pull request by t1m0thyj**
**What It Does**
Handles control characters in data set member names returned by the `List.allMembers` API.

The "esnext" lib in tsconfig.json is required to use the `matchAll` function.

**How to Test**

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: packages/zosfiles/src/methods/list/List.ts [hunk 1]
```diff
@@ -74,7 +74,28 @@ export class List {
...
+                // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+                for (let i = 1; i <= memberMaxLength; i++) {
+                    const memberChar = data.charCodeAt(memberStartIdx + i);
+                    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+                    if (memberChar < 32) {
+                        // Replace control characters with Unicode <?> symbol
+                        data = data.substring(0, memberStartIdx + i) + "\\ufffd" + data.substring(memberStartIdx + i + 1);
+                        memberStartIdx += 5;  // eslint-disable-line @typescript-eslint/no-magic-numbers
+                    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+                    } else if (memberChar === 34 && data.charCodeAt(memberStartIdx + i - 1) !== 92) {
```
Comment (zFernand0) [inline packages/zosfiles/src/methods/list/List.ts:92]: Can we add constants for these numbers? That may help devs understand what 34 and 92 are. :yum:

File: packages/zosfiles/src/methods/list/List.ts [hunk 2]
```diff
@@ -74,7 +74,22 @@
...
+            try {
+                response = JSONUtils.parse(data);
+            } catch {
+                // Escape invalid JSON characters in encrypted member names
+                for (const match of Array.from(data.matchAll(/"member":\s*"(?![A-Za-z@#$][A-Za-z0-9@#$]{0,7}")/g)).reverse()) {
+                    const memberStartIdx = match.index + match[0].length;
+                    const memberNameLength = 8;
+                    const memberName = data.substring(memberStartIdx, memberStartIdx + memberNameLength);
+                    // eslint-disable-next-line no-control-regex
+                    const escapedMemberName = memberName.replace(/[\x00-\x1f]/g, "\\ufffd").replace(/"/g, `\\"`);
```
[the remark this answers is not in the document]
Comment (t1m0thyj) [reply packages/zosfiles/src/methods/list/List.ts:88]: Is this something we need to fix or can it be dismissed? Maybe I'm missing something but I don't see a problem with this regex.

[the remark this answers is not in the document]
Comment (awharn) [reply packages/zosfiles/src/methods/list/List.ts:88]: I think we can close it - it is warning saying that there might be something in here if we are attempting to sanitize untrusted user input before throwing it into a command. We are just trying to find control characters in a string and replace those so we have valid JSON, which is very specifically targeted sanitization - and we aren't taking in user input, but an API response.

**Review by awharn — APPROVED** [review]

**Review by awharn — APPROVED** [review]

**Review by gejohnston — APPROVED** [review]
Looks good.

**Pull request merged**